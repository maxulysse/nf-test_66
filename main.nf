include { MD5SUM } from './modules/nf-core/md5sum/main.nf'

workflow MAIN {
    take:

    main:
    input = Channel.fromPath(params.test_data['sarscov2']['illumina']['test_1_fastq_gz']).map{ file -> [ [id:file.baseName], file ] }

    MD5SUM(input)

    MD5SUM.out.checksum.collectFile(storeDir: "${params.outdir}/csv") { meta, checksum ->
        md5sum = "${params.outdir}/${checksum.fileName}"

        ["md5sum.csv", "${md5sum}\n"]
    }
}

workflow {

    MAIN()

}