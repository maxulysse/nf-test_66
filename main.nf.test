nextflow_workflow {

    name "Test workflow in main.nf"
    script "main.nf"
    workflow "MAIN"

    test("MAIN: default test") {
        when {
            workflow {
                """
                """

            }
            params {
                outdir = "$outputDir"
            }
        }
        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(path("$outputDir/").list()).match("outputPath") }
            )
        }
    }
}