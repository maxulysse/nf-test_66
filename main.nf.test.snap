{
    "outputPath": {
        "content": [
            [
                "md5sum.csv:md5,542e45a543992b87a2ae215f00d40b3e"
            ],
            "test_1.fastq.gz.md5:md5,63a3f187d06ae5cffe77fcd39bbdd63a"
        ],
        "timestamp": "2023-08-29T17:04:22+0000"
    },
    "MAIN: default test": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_2.fastq"
                        },
                        "test_2.fastq.gz.md5:md5,055dc46636836dbd1d9ede2e6ce8cd4e"
                    ],
                    [
                        {
                            "id": "test_1.fastq"
                        },
                        "test_1.fastq.gz.md5:md5,63a3f187d06ae5cffe77fcd39bbdd63a"
                    ],
                    [
                        {
                            "id": "test_pair",
                            "single_end": false
                        },
                        [
                            "test_pair_1_fastqc.zip:md5,e96ad8acd2be79e846b326563c24b5fa",
                            "test_pair_2_fastqc.zip:md5,76a47ffa2e4b1a1828b1b3ada20490e7"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,34a4c33329b18e35befc60b7ce54d5f1",
                    "versions.yml:md5,62f68e994c87e146a7af55d5b78a0ce7"
                ],
                "expected_files": [
                    [
                        {
                            "id": "test_2.fastq"
                        },
                        "test_2.fastq.gz.md5:md5,055dc46636836dbd1d9ede2e6ce8cd4e"
                    ],
                    [
                        {
                            "id": "test_1.fastq"
                        },
                        "test_1.fastq.gz.md5:md5,63a3f187d06ae5cffe77fcd39bbdd63a"
                    ],
                    [
                        {
                            "id": "test_pair",
                            "single_end": false
                        },
                        [
                            "test_pair_1_fastqc.zip:md5,e96ad8acd2be79e846b326563c24b5fa",
                            "test_pair_2_fastqc.zip:md5,76a47ffa2e4b1a1828b1b3ada20490e7"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,34a4c33329b18e35befc60b7ce54d5f1",
                    "versions.yml:md5,62f68e994c87e146a7af55d5b78a0ce7"
                ]
            }
        ],
        "timestamp": "2023-08-29T17:04:22+0000"
    }
}